bench/main.go

```diff
@@ -40,10 +41,10 @@ func main() {
 
 	mm := newMemMetrics()
 	ba := getBackend(*b, backend.WithLogger(slog.New(&nullHandler{})), backend.WithMetrics(mm))
-
+	mpb := monoprocess.NewMonoprocessBackend(ba)
```

Comment:
Could we move this down to `getBackend` and only use it for mysql and sqlite? Probably won't hurt for Redis, but since it's not really required there I think we should benchmark without.

Reply:
Isn't the poll logic part of `ActivityWorker` and thus would affect any backend implementation?

Reply:
No that should be fine. The redis backend uses [blocking stream reads](https://developer.redis.com/develop/dotnet/streams/blocking-reads/) and waits some time for new workflow/activity tasks to be ready to pickup.

Reply:
@cschleiden I pushed the changes. As you can see from this test the `memory` backend also needed it

```
vroldanbet@vrb-mbp bench % hyperfine --warmup 1 -m 50 'go run .  -runs 2 -backend sqlite'                                                                                          
Benchmark 1: go run .  -runs 2 -backend sqlite
  Time (mean ± σ):     722.5 ms ±  36.7 ms    [User: 498.7 ms, System: 451.4 ms]
  Range (min … max):   653.5 ms … 789.6 ms    50 runs
 
vroldanbet@vrb-mbp bench % hyperfine --warmup 1 -m 50 'go run .  -runs 2 -backend memory'
Benchmark 1: go run .  -runs 2 -backend memory
  Time (mean ± σ):      1.290 s ±  0.494 s    [User: 0.543 s, System: 0.429 s]
  Range (min … max):    0.519 s …  2.191 s    50 runs
 
vroldanbet@vrb-mbp bench % hyperfine --warmup 1 -m 50 'go run .  -runs 2 -backend memory'
Benchmark 1: go run .  -runs 2 -backend memory
  Time (mean ± σ):     540.5 ms ±  13.1 ms    [User: 495.9 ms, System: 394.9 ms]
  Range (min … max):   512.6 ms … 592.4 ms    50 runs
```

Let's run this in CI and see what the numbers look like